mmdet/evaluation/metrics/crowdhuman_metric.py

```diff
@@ -0,0 +1,814 @@
+                 eval_mode: int = 0,
+                 iou_thres: float = 0.5,
+                 compare_matching_method: Optional[str] = None,
+                 mr_ref: str = 'CALTECH_-2',
+                 ji_process_num: int = 10) -> None:
+        super().__init__(collect_device=collect_device, prefix=prefix)
+
+        self.ann_file = ann_file
+        # crowdhuman evaluation metrics
+        self.metrics = metric if isinstance(metric, list) else [metric]
+        allowed_metrics = ['MR', 'AP', 'JI']
+        for metric in self.metrics:
+            if metric not in allowed_metrics:
+                raise KeyError(f"metric should be one of 'MR', 'AP', 'JI',"
+                               f'but got {metric}.')
+
+        self.format_only = format_only
+        if self.format_only:
+            assert outfile_prefix is not None, 'outfile_prefix must be not'
+            'None when format_only is True, otherwise the result files will'
+            'be saved to a temp directory which will be cleaned up at the end.'
+        self.outfile_prefix = outfile_prefix
+
+        assert eval_mode in [0, 1, 2], \
+            "Unknown eval mode. mr_ref should be one of '0', '1', '2'."
+        assert compare_matching_method is None or \
+               compare_matching_method == 'VOC', \
+               'The alternative compare_matching_method is VOC.' \
+               'This parameter defaults to CALTECH(None)'
+        assert mr_ref == 'CALTECH_-2' or mr_ref == 'CALTECH_-4', \
+            "mr_ref should be one of 'CALTECH_-2', 'CALTECH_-4'."
+        self.eval_mode = eval_mode
+        self.iou_thres = iou_thres
+        self.compare_matching_method = compare_matching_method
+        self.mr_ref = mr_ref
+        self.ji_process_num = ji_process_num
+
+    @staticmethod
+    def results2json(results: Sequence[dict], outfile_prefix: str) -> str:
+        """Dump the detection results to a json file."""
+        result_file_path = f'{outfile_prefix}.json'
+        with open(result_file_path, 'w') as outfile:
+            for i, result in enumerate(results):
+                ann, pred = result
+                dump_dict = dict()
+                dump_dict['ID'] = ann['ID']
+                dump_dict['width'] = ann['width']
+                dump_dict['height'] = ann['height']
+                dtboxes = []
+                bboxes = pred.tolist()
+                for _, single_bbox in enumerate(bboxes):
+                    temp_dict = dict()
+                    x1, y1, x2, y2, score = single_bbox
+                    temp_dict['box'] = [x1, y1, x2 - x1, y2 - y1]
+                    temp_dict['score'] = score
+                    temp_dict['tag'] = 1
+                    dtboxes.append(temp_dict)
+                dump_dict['dtboxes'] = dtboxes
+                outfile.write(json.dumps(dump_dict) + '\n')
+
+        return result_file_path
+
+    def process(self, data_batch: Sequence[dict],
+                predictions: Sequence[dict]) -> None:
```

Comment:
MMEngine and MMDetection data flow is changed. Need to update the signature.
```suggestion
                data_samples: Sequence[dict]) -> None:
```